numpy/core/src/umath/ufunc_type_resolution.c

```diff
@@ -641,7 +641,7 @@ PyUFunc_SimpleUniformOperationTypeResolver(
     /* Check against the casting rules */
     if (PyUFunc_ValidateCasting(ufunc, casting, operands, out_dtypes) < 0) {
         for (int iop = 0; iop < nop; iop++) {
-            Py_DECREF(out_dtypes[iop]);
+            Py_XDECREF(out_dtypes[iop]);
```

Comment:
```suggestion
            Py_DECREF(out_dtypes[iop]);
```
I understand the intent, but I honestly think there is no point.  `ValidateCasting` would barf on this already if this actually happened.

Reply:
So maybe a comment would help future me understand.

Reply:
I don't understand. If one of the out_dtypes is NULL, `PyUFunc_ValidateCasting` will set an error and return -1. It would still loop through the deallocation.

Reply:
The `operands` can be NULL (for outputs), but `out_dtypes` cannot be?